lib/node_modules/@stdlib/array/base/count-ifs/docs/types/index.d.ts

```diff
@@ -0,0 +1,149 @@
+/*
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 4.1
+
+/// <reference types="@stdlib/types"/>
+
+import { Collection, AccessorArrayLike } from '@stdlib/types/array';
+
+/**
+* Returns a boolean indicating whether an element passes a test.
+*
+* @returns boolean indicating whether an element passes a test
+*/
+type Nullary = () => boolean;
+
+/**
+* Returns a boolean indicating whether an element passes a test.
+*
+* @param value - current array element
+* @returns boolean indicating whether an element passes a test
+*/
+type Unary<U> = ( value: U ) => boolean;
+
+/**
+* Returns a boolean indicating whether an element passes a test.
+*
+* @param value - current array element
+* @param index - current array element index
+* @returns boolean indicating whether an element passes a test
+*/
+type Binary<U> = ( value: U, index: number ) => boolean;
+
+/**
+* Returns a boolean indicating whether an element passes a test.
+*
+* @param value - current array element
+* @param index - current array element index
+* @param arr - input array
+* @returns boolean indicating whether an element passes a test
+*/
+type Ternary<U> = ( value: U, index: number, arr: Collection<U> | AccessorArrayLike<U> ) => boolean;
+
+/**
+* Returns a boolean indicating whether an element passes a test.
+*
+* @param value - current array element
+* @param index - current array element index
+* @param arr - input array
+* @returns boolean indicating whether an element passes a test
+*/
+type Predicate<U> = Nullary | Unary<U> | Binary<U> | Ternary<U>;
+
+/**
+* Counts the number of elements in the input arrays which pass the tests implemented by the predicate functions.
+*
+* @param x - input array
+* @param predicate - predicate function
+* @returns result
+*
+* @example
+* function predicate( v ) {
+*     return v > 0;
+* }
+*
+* var x = [ 0, 1, 0, 1, 1 ];
+*
+* var n = countIfs( x, predicate );
+* // returns 3
+*/
+declare function countIfs<U = unknown>( x: Collection<U> | AccessorArrayLike<U>, predicate: Predicate<U> ): number;
```

Comment:
Create a type `InputArray<U> = Collection<U> | AccessorArrayLike<U>`. Then, see about doing

```
declare function countIfs<T = unknown, U extends InputArray<T>>( x: U, predicate: Predicate<T, U> ): number;
```

You'll then need to update the predicates accordingly. In short, we want to preserve type information for the `arr` argument. Otherwise, right now, we lose type specificity in the callback.